modules/library/main/src/main/java/org/geotools/filter/function/AndFunction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2002-2008, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ */
+package org.geotools.filter.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+/**
+ * This function is used to evaluate a And operator with a left and right operands
+ *
+ * @author Erwan Bocher, CNRS, 2020
+ */
+public class AndFunction extends FunctionExpressionImpl {
+
+    public static FunctionName NAME =
+            new FunctionNameImpl(
+                    "And",
+                    Boolean.class,
+                    parameter("left", Boolean.class),
```

Comment:
I think you commented on it already, and it's just for talking, not a request, but ... what about having a varying number of inputs, allow the user to pass as many arguments as they want? Should make setting up complex expressions easier. 
In case you're interested, the "in" function can be used as a template to implement this behavior:

https://github.com/geotools/geotools/blob/41ac131a56b020bd6bf1436e8fd0fa303f7927d1/modules/library/main/src/main/java/org/geotools/filter/function/InFunction.java#L77

I'm happy to merge as it is now though, as said, not a requirement.

Reply:
I can check the In logic. Can you point me the resource ?

Reply:
The previous comment contains a link to it, follow it, it's the entire InFunction:

https://github.com/geotools/geotools/blob/41ac131a56b020bd6bf1436e8fd0fa303f7927d1/modules/library/main/src/main/java/org/geotools/filter/function/InFunction.java#L77

Reply:
Ok it seems InFunction ;-)
By the way I have a question.
What about  this expression
type = 'geotools' or version > 20 and language = 'it'

Is geotools internal function evaluator is able to manage precedence between  or and and op ?
or should I build a  tree to manage that ?

Reply:
Functions do not have precedence, they are applied in the order they are called.

Reply:
So after looking on Function mechanism, I think you can merge this PR. The setting up of the function precedence should be left to the developer's appreciation.